types/sjcl/index.d.ts

```diff
@@ -189,7 +189,7 @@ declare namespace sjcl {
     // ________________________________________________________________________
 
     interface SjclCodec<T> {
-        fromBits(bits: BitArray): T;
+        fromBits(bits: BitArray, _noEquals?: boolean, _url?: boolean): T;
         toBits(value: T): BitArray;
```

Comment:
The only codecs that use these variables are `core/codecBase32.js` and `core/codecBase64.js`.
The definition of `toBits` is also different (`toBits(value: T, _url: boolean): BitArray`).

I think a better approach would be to instead create an interface for base32 and base64 codecs that extends SjclCodec<string> and include these variables there, e.g.:
```
interface SjclBase32Codec extends SjclCodec<string> {
    fromBits(bits: BitArray, _noEquals?: boolean, _hex?: boolean): string;
    toBits(value: string, _hex?: boolean): BitArray;
}

interface SjclBase64Codec extends SjclCodec<string> {
    fromBits(bits: BitArray, _noEquals?: boolean, _url?: boolean): string;
    toBits(value: string, _url?: boolean): BitArray;
}
```

There is also `base32hex`, which seems to include `_noEquals`, but omits `_hex`.

Reply:
Hi, didn't have time to work on this.
I think a better approach is to not relying so much on abstraction. i.e. don't force DRY.